server/service.go

```diff
@@ -34,87 +32,50 @@ func newService(svcInfo *serviceinfo.ServiceInfo, handler interface{}) *service
+		svcMap:               map[string]*service{},
+		conflictingMethodMap: map[string]bool{},
 	}
 }
 
 func (s *services) addService(svcInfo *serviceinfo.ServiceInfo, handler interface{}, registerOpts *RegisterOptions) error {
 	svc := newService(svcInfo, handler)
-
-	if err := s.checkCombineServiceWithOtherService(svcInfo); err != nil {
-		return err
-	}
-
-	if err := s.checkMultipleFallbackService(registerOpts, svc); err != nil {
-		return err
-	}
-
-	s.svcMap[svcInfo.ServiceName] = svc
-	s.createSearchMap(svcInfo, svc, registerOpts)
-	return nil
-}
-
-// when registering combine service, it does not allow the registration of other services
-func (s *services) checkCombineServiceWithOtherService(svcInfo *serviceinfo.ServiceInfo) error {
-	if len(s.svcMap) > 0 {
-		if _, ok := s.svcMap["CombineService"]; ok || svcInfo.ServiceName == "CombineService" {
-			return errors.New("only one service can be registered when registering combine service")
-		}
-	}
-	return nil
-}
-
-func (s *services) checkMultipleFallbackService(registerOpts *RegisterOptions, svc *service) error {
 	if registerOpts.IsFallbackService {
 		if s.fallbackSvc != nil {
 			return fmt.Errorf("multiple fallback services cannot be registered. [%s] is already registered as a fallback service", s.fallbackSvc.svcInfo.ServiceName)
 		}
 		s.fallbackSvc = svc
 	}
-	return nil
-}
-
-func (s *services) createSearchMap(svcInfo *serviceinfo.ServiceInfo, svc *service, registerOpts *RegisterOptions) {
+	s.svcMap[svcInfo.ServiceName] = svc
+	// method search map
 	for methodName := range svcInfo.Methods {
-		s.svcSearchMap[remote.BuildMultiServiceKey(svcInfo.ServiceName, methodName)] = svc
-		if svcFromMap, ok := s.svcSearchMap[methodName]; ok {
-			s.handleConflictingMethod(svcFromMap, svc, methodName, registerOpts)
+		if _, ok := s.methodSvcMap[methodName]; ok {
+			s.handleConflictingMethod(svc, methodName, registerOpts)
 		} else {
-			s.svcSearchMap[methodName] = svc
+			s.methodSvcMap[methodName] = svc
 		}
 	}
+	return nil
 }
 
-func (s *services) handleConflictingMethod(svcFromMap, svc *service, methodName string, registerOpts *RegisterOptions) {
-	s.registerConflictingMethodHasFallbackSvcMap(svcFromMap, methodName)
-	s.updateWithFallbackSvc(registerOpts, svc, methodName)
-}
-
-func (s *services) registerConflictingMethodHasFallbackSvcMap(svcFromMap *service, methodName string) {
-	if _, ok := s.conflictingMethodHasFallbackSvcMap[methodName]; !ok {
-		if s.fallbackSvc != nil && svcFromMap.svcInfo.ServiceName == s.fallbackSvc.svcInfo.ServiceName {
-			// svc which is already registered is a fallback service
-			s.conflictingMethodHasFallbackSvcMap[methodName] = true
-		} else {
-			s.conflictingMethodHasFallbackSvcMap[methodName] = false
-		}
+func (s *services) handleConflictingMethod(svc *service, methodName string, registerOpts *RegisterOptions) {
+	// true means has conflicting method
+	if _, ok := s.conflictingMethodMap[methodName]; !ok {
```

Comment:
The method was not originally registered but was assigned a value of true, it doesn't make sense.